core/trino-main/src/main/java/io/trino/operator/SplitOperatorInfo.java

```diff
@@ -53,4 +57,22 @@ public CatalogHandle getCatalogHandle()
     {
         return catalogHandle;
     }
+
+    @Override
+    public SplitOperatorInfo mergeWith(SplitOperatorInfo other)
+    {
+        return mergeWith(List.of(other));
+    }
+
+    @Override
+    public SplitOperatorInfo mergeWith(List<SplitOperatorInfo> others)
+    {
+        return new SplitOperatorInfo(
+                catalogHandle,
+                mergeMaps(
+                        Stream.concat(
+                                Stream.of(splitInfo),
+                                others.stream().map(SplitOperatorInfo::getSplitInfo)),
+                        (first, second) -> String.join(", ", first, second)));
```

Comment:
I think this is going to be potentially slow for scans with a large number of splits and might end up producing large strings. Ideally, the `io.trino.spi.connector.ConnectorSplit#getSplitInfo` API should be fixed to return a `Metrics` object instead but that will trigger a lot of changes as every connector's implementation will need to be updated.
Instead of merging maps just take the current map and add " (" + others.size() + " more)" to every map value.
Your test can be updated to check that the unload directory for the query is empty/deleted instead of checking each location.

Reply:
added suggested values in the map.